src/olympia/abuse/serializers.py

```diff
@@ -1,16 +1,79 @@
 from rest_framework import serializers
 
+from olympia import amo
 from olympia.abuse.models import AbuseReport
 from olympia.accounts.serializers import BaseUserSerializer
+from olympia.api.fields import ReverseChoiceField
 
 
-class AddonAbuseReportSerializer(serializers.ModelSerializer):
-    addon = serializers.SerializerMethodField()
+class BaseAbuseReportSerializer(serializers.ModelSerializer):
     reporter = BaseUserSerializer(read_only=True)
 
     class Meta:
         model = AbuseReport
-        fields = ('reporter', 'addon', 'message')
+        fields = ('message', 'reporter')
+
+    def validate_target(self, data, target_name):
+        if target_name not in data:
+            msg = serializers.Field.default_error_messages['required']
+            raise serializers.ValidationError({
+                target_name: [msg]
+            })
+
+    def to_internal_value(self, data):
+        output = super(BaseAbuseReportSerializer, self).to_internal_value(data)
+        request = self.context['request']
+        output['ip_address'] = request.META.get('REMOTE_ADDR')
```

Comment:
We should have `ip_address` defined in the serializer so that it gets validated and we don't store unvalidated input.

Reply:
The problem is we don't want it to be submitted directly. And we don't want to error on it either. So I think it's the best compromise. It will go away in https://github.com/mozilla/addons-server/issues/10720 anyway.

Reply:
Ah, if it goes away (I forgot) that's fine then.